devdox/run_migrations.py

```
@@ -6,11 +6,308 @@
+    bmap = _index_by(before, keycols)
+    amap = _index_by(after, keycols)
+    added_keys   = list(amap.keys() - bmap.keys())
+    removed_keys = list(bmap.keys() - amap.keys())
+    common_keys  = list(amap.keys() & bmap.keys())
+
+    added   = [amap[k] for k in sorted(added_keys)]
+    removed = [bmap[k] for k in sorted(removed_keys)]
+    changed = []
+
+    for k in sorted(common_keys):
+        b = bmap[k]; a = amap[k]
```

Ruff (0.13.1)

Multiple statements on one line (semicolon)

(E702)